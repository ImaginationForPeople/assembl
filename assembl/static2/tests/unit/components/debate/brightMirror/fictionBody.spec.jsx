// @flow
import React from 'react';
/* eslint-disable import/no-extraneous-dependencies */
import initStoryshots from '@storybook/addon-storyshots';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
/* eslint-enable */

import FictionBody from '../../../../../js/app/components/debate/brightMirror/fictionBody';
import type { FictionBodyType } from '../../../../../js/app/components/debate/brightMirror/fictionBody';

// Separate the snapshots in directories next to each component
// Name should match with the story name
initStoryshots({
  storyKindRegex: /^FictionBody$/
});

configure({ adapter: new Adapter() });

const defaultTitle: string = 'Fugit veritatis nemo';
const defaultContent: string = 'Vero et ut et quia quo. Molestiae ut cupiditate odio numquam veniam esse cumque modi.';

const defaultFictionBody: FictionBodyType = {
  title: defaultTitle,
  content: defaultContent
};

describe('<FictionBody /> - with shallow', () => {
  let wrapper;
  let fictionBody: FictionBodyType;

  beforeEach(() => {
    fictionBody = { ...defaultFictionBody };
    wrapper = shallow(<FictionBody {...fictionBody} />);
  });

  it('should render one h1 title and one div content', () => {
    expect(wrapper.find('h1 [className="fiction-title"]')).toHaveLength(1);
    expect(wrapper.find('div [className="fiction-content"]')).toHaveLength(1);
  });

  it('should display the fiction title', () => {
    expect(wrapper.contains(defaultTitle)).toBe(true);
  });

  it('should display "no title specified" when title is set to null', () => {
    wrapper.setProps({ title: '' });
    expect(wrapper.contains('no title specified')).toBe(true);
  });
});

describe('<FictionBody /> - with mount', () => {
  let wrapper;
  let fictionBody: FictionBodyType;

  beforeEach(() => {
    fictionBody = { ...defaultFictionBody };
    wrapper = mount(<FictionBody {...fictionBody} />);
  });

  it('should display the fiction content', () => {
    const fictionContent: string = wrapper.find('div [className="fiction-content"]').text();
    expect(fictionContent).toEqual(defaultContent);
  });

  it('should display "no content specified" when content is set to null', () => {
    wrapper.setProps({ content: '' });
    const fictionContent: string = wrapper.find('div [className="fiction-content"]').text();
    expect(fictionContent).toEqual('no content specified');
  });
});