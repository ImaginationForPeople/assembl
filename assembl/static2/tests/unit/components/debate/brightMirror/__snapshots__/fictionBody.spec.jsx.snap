// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots FictionBody default 1`] = `
<div>
  <div
    style={
      Object {
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <h1
      className="fiction-title"
    >
      Les émotifs
    </h1>
    <div
      className="fiction-content"
    >
      
          &lt;p&gt;15 janvier 2050 : je suis assis paisiblement chez moi en train de lire un épisode de la série « Ceux qui restent »
          en version émotive. L’histoire me rend triste mais je la lis parce que j’y entrevois la profondeur des sentiments.
          Avant l’arrivée de la tablette sensorielle, quand les films et les séries n’avaient pas d’émo-description, je
          préférais les histoires écrites. Au moins dans les livres, les sentiments sont détaillés. Je les comprends. Je
          suis là, dans l’histoire, dans la vie des personnages, dans leur tête, dans leur âme. J’entre en introspection
          // pour sonder puis ranger dans les tiroirs de mon esprit les nouvelles émotions que je découvre.&lt;/p&gt;
      
          &lt;p&gt;Maintenant, avec les sous-titres c’est plus facile pour moi de regarder une vidéo : « Confusion indice 5 + Honte
          indice 1 » ; ou « Joie indice 3 + Nostalgie indice 2 » ; ou encore « Colère indice 8 + Amour indice 9 + Regret
          indice 4 » Ces idéogrammes qui apparaissent à chaque scène que je parcours du doigt, je sais les lire couramment
          maintenant. C’est comme du braille digital, mais pour les gens que le monde des interactions déroute. C’est généré
          par une IA qui analyse le format facial des inférences émotionnelles pour les traduire en émo-signes qui se
          dessinent et se déploient comme des fleurs sur l’écran. En lisant, je prends en moi la palette des émotions qui
          existent. Et j’apprends à lier le signe au sentiment. Cela me donne une sensation d’universalité qui se manifeste
          presque de façon physique. Comme un courant électrique qui me traverse. Très intime, personnel. Mais tellement
          collectif : je me sens comme tout le monde. J’appartiens moi aussi. Je peux entrer en interférence émotionnelle
          avec un univers de sensations, toucher le monde des impressions et des consciences humaines.&lt;/p&gt;
        
    </div>
  </div>
  <button
    onClick={[Function]}
    style={
      Object {
        "background": "#28c",
        "border": "none",
        "borderRadius": "0 0 0 5px",
        "color": "#fff",
        "cursor": "pointer",
        "display": "block",
        "fontFamily": "sans-serif",
        "fontSize": "12px",
        "padding": "5px 15px",
        "position": "fixed",
        "right": 0,
        "top": 0,
      }
    }
    type="button"
  >
    Show Info
  </button>
  <div
    style={
      Object {
        "background": "white",
        "bottom": 0,
        "display": "none",
        "left": 0,
        "overflow": "auto",
        "padding": "0 40px",
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 99999,
      }
    }
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "#28c",
          "border": "none",
          "borderRadius": "0 0 0 5px",
          "color": "#fff",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "sans-serif",
          "fontSize": "12px",
          "padding": "5px 15px",
          "position": "fixed",
          "right": 0,
          "top": 0,
        }
      }
      type="button"
    >
      ×
    </button>
    <div
      style={undefined}
    >
      <div
        style={
          Object {
            "WebkitFontSmoothing": "antialiased",
            "backgroundColor": "#fff",
            "border": "1px solid #eee",
            "borderRadius": "2px",
            "boxShadow": "0px 2px 3px rgba(0, 0, 0, 0.05)",
            "color": "#444",
            "fontFamily": "-apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", \\"Arial\\", sans-serif",
            "fontSize": "15px",
            "fontWeight": 300,
            "lineHeight": 1.45,
            "marginBottom": "20px",
            "marginTop": "20px",
            "padding": "20px 40px 40px",
          }
        }
      >
        <div
          style={
            Object {
              "borderBottom": "1px solid #eee",
              "marginBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <h1
            style={
              Object {
                "fontSize": "35px",
                "margin": 0,
                "padding": 0,
              }
            }
          >
            FictionBody
          </h1>
          <h2
            style={
              Object {
                "fontSize": "22px",
                "fontWeight": 400,
                "margin": "0 0 10px 0",
                "padding": 0,
              }
            }
          >
            default
          </h2>
        </div>
        
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Story Source
          </h1>
          <pre
            className="css-4akams"
          >
            <div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &lt;
                  fictionBody
                </span>
                <span>
                  <span>
                     
                    <span
                      style={Object {}}
                    >
                      title
                    </span>
                    <span>
                      =
                      <span
                        style={Object {}}
                      >
                        "
                        <span
                          style={
                            Object {
                              "color": "#22a",
                              "wordBreak": "break-word",
                            }
                          }
                        >
                          Les émotifs
                        </span>
                        "
                      </span>
                    </span>
                  </span>
                  <span>
                     
                    <span
                      style={Object {}}
                    >
                      content
                    </span>
                    <span>
                      =
                      <span
                        style={Object {}}
                      >
                        "
                        <span
                          style={
                            Object {
                              "color": "#22a",
                              "wordBreak": "break-word",
                            }
                          }
                        >
                          
                              &lt;p&gt;15 janvier 2050 : je suis assis paisibleme…
                        </span>
                        "
                      </span>
                    </span>
                     
                  </span>
                </span>
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  /&gt;
                </span>
              </div>
            </div>
            <button
              className="css-py33tc"
              onClick={[Function]}
            >
              <div
                className="css-kv47nt"
              >
                <div
                  style={
                    Object {
                      "marginBottom": 6,
                    }
                  }
                >
                  Copied!
                </div>
                <div>
                  Copy
                </div>
              </div>
            </button>
          </pre>
        </div>
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Prop Types
          </h1>
          <div>
            <h2
              style={
                Object {
                  "margin": "20px 0 0 0",
                }
              }
            >
              "
              fictionBody
              " Component
            </h2>
            <small>
              No propTypes defined!
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots FictionBody no content  1`] = `
<div>
  <div
    style={
      Object {
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <h1
      className="fiction-title"
    >
      no title specified
    </h1>
    <div
      className="fiction-content"
    >
      no content specified
    </div>
  </div>
  <button
    onClick={[Function]}
    style={
      Object {
        "background": "#28c",
        "border": "none",
        "borderRadius": "0 0 0 5px",
        "color": "#fff",
        "cursor": "pointer",
        "display": "block",
        "fontFamily": "sans-serif",
        "fontSize": "12px",
        "padding": "5px 15px",
        "position": "fixed",
        "right": 0,
        "top": 0,
      }
    }
    type="button"
  >
    Show Info
  </button>
  <div
    style={
      Object {
        "background": "white",
        "bottom": 0,
        "display": "none",
        "left": 0,
        "overflow": "auto",
        "padding": "0 40px",
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 99999,
      }
    }
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "#28c",
          "border": "none",
          "borderRadius": "0 0 0 5px",
          "color": "#fff",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "sans-serif",
          "fontSize": "12px",
          "padding": "5px 15px",
          "position": "fixed",
          "right": 0,
          "top": 0,
        }
      }
      type="button"
    >
      ×
    </button>
    <div
      style={undefined}
    >
      <div
        style={
          Object {
            "WebkitFontSmoothing": "antialiased",
            "backgroundColor": "#fff",
            "border": "1px solid #eee",
            "borderRadius": "2px",
            "boxShadow": "0px 2px 3px rgba(0, 0, 0, 0.05)",
            "color": "#444",
            "fontFamily": "-apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", \\"Arial\\", sans-serif",
            "fontSize": "15px",
            "fontWeight": 300,
            "lineHeight": 1.45,
            "marginBottom": "20px",
            "marginTop": "20px",
            "padding": "20px 40px 40px",
          }
        }
      >
        <div
          style={
            Object {
              "borderBottom": "1px solid #eee",
              "marginBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <h1
            style={
              Object {
                "fontSize": "35px",
                "margin": 0,
                "padding": 0,
              }
            }
          >
            FictionBody
          </h1>
          <h2
            style={
              Object {
                "fontSize": "22px",
                "fontWeight": 400,
                "margin": "0 0 10px 0",
                "padding": 0,
              }
            }
          >
            no content 
          </h2>
        </div>
        
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Story Source
          </h1>
          <pre
            className="css-4akams"
          >
            <div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &lt;
                  fictionBody
                </span>
                <span>
                  <span>
                     
                    <span
                      style={Object {}}
                    >
                      title
                    </span>
                    <span>
                      =
                      <span
                        style={Object {}}
                      >
                        "
                        <span
                          style={
                            Object {
                              "color": "#22a",
                              "wordBreak": "break-word",
                            }
                          }
                        >
                          
                        </span>
                        "
                      </span>
                    </span>
                  </span>
                  <span>
                     
                    <span
                      style={Object {}}
                    >
                      content
                    </span>
                    <span>
                      =
                      <span
                        style={Object {}}
                      >
                        "
                        <span
                          style={
                            Object {
                              "color": "#22a",
                              "wordBreak": "break-word",
                            }
                          }
                        >
                          
                        </span>
                        "
                      </span>
                    </span>
                     
                  </span>
                </span>
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  /&gt;
                </span>
              </div>
            </div>
            <button
              className="css-py33tc"
              onClick={[Function]}
            >
              <div
                className="css-kv47nt"
              >
                <div
                  style={
                    Object {
                      "marginBottom": 6,
                    }
                  }
                >
                  Copied!
                </div>
                <div>
                  Copy
                </div>
              </div>
            </button>
          </pre>
        </div>
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Prop Types
          </h1>
          <div>
            <h2
              style={
                Object {
                  "margin": "20px 0 0 0",
                }
              }
            >
              "
              fictionBody
              " Component
            </h2>
            <small>
              No propTypes defined!
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots FictionBody playground 1`] = `
<div>
  <div
    style={
      Object {
        "position": "relative",
        "zIndex": 0,
      }
    }
  >
    <h1
      className="fiction-title"
    >
      Les émotifs
    </h1>
    <div
      className="fiction-content"
    >
      
          &lt;p&gt;15 janvier 2050 : je suis assis paisiblement chez moi en train de lire un épisode de la série « Ceux qui restent »
          en version émotive. L’histoire me rend triste mais je la lis parce que j’y entrevois la profondeur des sentiments.
          Avant l’arrivée de la tablette sensorielle, quand les films et les séries n’avaient pas d’émo-description, je
          préférais les histoires écrites. Au moins dans les livres, les sentiments sont détaillés. Je les comprends. Je
          suis là, dans l’histoire, dans la vie des personnages, dans leur tête, dans leur âme. J’entre en introspection
          // pour sonder puis ranger dans les tiroirs de mon esprit les nouvelles émotions que je découvre.&lt;/p&gt;
      
          &lt;p&gt;Maintenant, avec les sous-titres c’est plus facile pour moi de regarder une vidéo : « Confusion indice 5 + Honte
          indice 1 » ; ou « Joie indice 3 + Nostalgie indice 2 » ; ou encore « Colère indice 8 + Amour indice 9 + Regret
          indice 4 » Ces idéogrammes qui apparaissent à chaque scène que je parcours du doigt, je sais les lire couramment
          maintenant. C’est comme du braille digital, mais pour les gens que le monde des interactions déroute. C’est généré
          par une IA qui analyse le format facial des inférences émotionnelles pour les traduire en émo-signes qui se
          dessinent et se déploient comme des fleurs sur l’écran. En lisant, je prends en moi la palette des émotions qui
          existent. Et j’apprends à lier le signe au sentiment. Cela me donne une sensation d’universalité qui se manifeste
          presque de façon physique. Comme un courant électrique qui me traverse. Très intime, personnel. Mais tellement
          collectif : je me sens comme tout le monde. J’appartiens moi aussi. Je peux entrer en interférence émotionnelle
          avec un univers de sensations, toucher le monde des impressions et des consciences humaines.&lt;/p&gt;
        
    </div>
  </div>
  <button
    onClick={[Function]}
    style={
      Object {
        "background": "#28c",
        "border": "none",
        "borderRadius": "0 0 0 5px",
        "color": "#fff",
        "cursor": "pointer",
        "display": "block",
        "fontFamily": "sans-serif",
        "fontSize": "12px",
        "padding": "5px 15px",
        "position": "fixed",
        "right": 0,
        "top": 0,
      }
    }
    type="button"
  >
    Show Info
  </button>
  <div
    style={
      Object {
        "background": "white",
        "bottom": 0,
        "display": "none",
        "left": 0,
        "overflow": "auto",
        "padding": "0 40px",
        "position": "fixed",
        "right": 0,
        "top": 0,
        "zIndex": 99999,
      }
    }
  >
    <button
      onClick={[Function]}
      style={
        Object {
          "background": "#28c",
          "border": "none",
          "borderRadius": "0 0 0 5px",
          "color": "#fff",
          "cursor": "pointer",
          "display": "block",
          "fontFamily": "sans-serif",
          "fontSize": "12px",
          "padding": "5px 15px",
          "position": "fixed",
          "right": 0,
          "top": 0,
        }
      }
      type="button"
    >
      ×
    </button>
    <div
      style={undefined}
    >
      <div
        style={
          Object {
            "WebkitFontSmoothing": "antialiased",
            "backgroundColor": "#fff",
            "border": "1px solid #eee",
            "borderRadius": "2px",
            "boxShadow": "0px 2px 3px rgba(0, 0, 0, 0.05)",
            "color": "#444",
            "fontFamily": "-apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Oxygen\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", \\"Arial\\", sans-serif",
            "fontSize": "15px",
            "fontWeight": 300,
            "lineHeight": 1.45,
            "marginBottom": "20px",
            "marginTop": "20px",
            "padding": "20px 40px 40px",
          }
        }
      >
        <div
          style={
            Object {
              "borderBottom": "1px solid #eee",
              "marginBottom": 10,
              "paddingTop": 10,
            }
          }
        >
          <h1
            style={
              Object {
                "fontSize": "35px",
                "margin": 0,
                "padding": 0,
              }
            }
          >
            FictionBody
          </h1>
          <h2
            style={
              Object {
                "fontSize": "22px",
                "fontWeight": 400,
                "margin": "0 0 10px 0",
                "padding": 0,
              }
            }
          >
            playground
          </h2>
        </div>
        
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Story Source
          </h1>
          <pre
            className="css-4akams"
          >
            <div>
              <div
                style={
                  Object {
                    "paddingLeft": 18,
                    "paddingRight": 3,
                  }
                }
              >
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  &lt;
                  fictionBody
                </span>
                <span>
                  <span>
                     
                    <span
                      style={Object {}}
                    >
                      title
                    </span>
                    <span>
                      =
                      <span
                        style={Object {}}
                      >
                        "
                        <span
                          style={
                            Object {
                              "color": "#22a",
                              "wordBreak": "break-word",
                            }
                          }
                        >
                          Les émotifs
                        </span>
                        "
                      </span>
                    </span>
                  </span>
                  <span>
                     
                    <span
                      style={Object {}}
                    >
                      content
                    </span>
                    <span>
                      =
                      <span
                        style={Object {}}
                      >
                        "
                        <span
                          style={
                            Object {
                              "color": "#22a",
                              "wordBreak": "break-word",
                            }
                          }
                        >
                          
                              &lt;p&gt;15 janvier 2050 : je suis assis paisibleme…
                        </span>
                        "
                      </span>
                    </span>
                     
                  </span>
                </span>
                <span
                  style={
                    Object {
                      "color": "#777",
                    }
                  }
                >
                  /&gt;
                </span>
              </div>
            </div>
            <button
              className="css-py33tc"
              onClick={[Function]}
            >
              <div
                className="css-kv47nt"
              >
                <div
                  style={
                    Object {
                      "marginBottom": 6,
                    }
                  }
                >
                  Copied!
                </div>
                <div>
                  Copy
                </div>
              </div>
            </button>
          </pre>
        </div>
        <div>
          <h1
            style={
              Object {
                "borderBottom": "1px solid #EEE",
                "fontSize": "25px",
                "margin": "20px 0 0 0",
                "padding": "0 0 5px 0",
              }
            }
          >
            Prop Types
          </h1>
          <div>
            <h2
              style={
                Object {
                  "margin": "20px 0 0 0",
                }
              }
            >
              "
              fictionBody
              " Component
            </h2>
            <small>
              No propTypes defined!
            </small>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
